src/App.tsx

```diff
@@ -1,14 +1,52 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+export enum Filters {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState(Filters.All);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(setTodos)
+      .finally(() => setLoading(false));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos
+      .filter(todo => {
+        if (filter === Filters.Active) {
+          return !todo.completed;
+        }
+
+        if (filter === Filters.Completed) {
+          return todo.completed;
+        }
+
+        return true;
+      })
+      .filter(todo =>
+        todo.title.trim().toLowerCase().includes(query.trim().toLowerCase()),
+      );
```

Comment:
Again filtering two times, just add your second filter logic to first, and it`s will be done